### PR TITLE
Correcting technical summary table for <summary>

### LINKED ISSUE #28446: Implicit role for `<summary>` element
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary

### What specific section or headline is this issue about?

[Technical summary](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#technical_summary)

### What information was incorrect, unhelpful, or incomplete?

The implicit ARIA role for the `<summary>` element is `button`

### What did you expect to see?

I don't know, the spec says that the implicit role is supposed to be `generic` but that many browsers use `button` instead. But chrome uses `generic` and firefox uses `summary`.
Maybe it would be less confusing (given the already very confusing situation) to have a specific message or something ?
Something like what is in the spec : "generic, note that many agents will use the button role instead"

### Do you have any supporting links, references, or citations?

> [No corresponding role](https://w3c.github.io/html-aria/#dfn-no-corresponding-role)
NOTE
Many, but not all, user agents expose the summary element with an implicit ARIA role=[button](https://w3c.github.io/html-aria/#index-aria-button).

-- [specification of the `<summary>` element](https://w3c.github.io/html-aria/#el-summary)

<div>
<figure>
  
  <figcaption>Chrome uses `generic` in the form of an implementation specific role `DisclosureTriangle`</figcaption>
</figure>
</div>

<div>
<figure>
  
  <figcaption>Firefox uses `summary` which is not documented on MDN</figcaption>
</figure>
</div>

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/summary`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/summary/index.md
* Last commit: https://github.com/mdn/content/commit/942a529383ee7ee3996fb234187641c08935f3ff
* Document last modified: 2023-04-13T06:13:15.000Z

</details>

### PR DESCRIPTION
**Pull request by Brian-Pob**
### Description

Correcting the implicit ARIA role in the technical summary table for the `summary` element.

### Motivation

To ensure the MDN docs follow the relevant HTML specs.

### Additional details

User [scottaohara](https://github.com/scottaohara) is an editor for the HTML AAM spec and the ARIA in HTML spec. He states in the linked issue that the table should:
1. Be updated to reference what the ARIA in HTML spec says
2. Include the ARIA in HTML spec in the Specifications section

### Related issues and pull requests

Fixes #28446

### CODE REVIEW

**Comment by Brian-Pob** [comment]
@scottaohara apologies for the @, I was hoping to get your insight on this. Thank you.

The MDN doc gives the following code as an example for using `<summary>` as a heading. [Summaries as Headings](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#summaries_as_headings)

```HTML
<details open>
  <summary><h4>Overview</h4></summary>
  <ol>
    <li>Cash on hand: $500.00</li>
    <li>Current invoice: $75.30</li>
    <li>Due date: 5/6/19</li>
  </ol>
</details>
```

The doc currently has a warning following the example:

> Warning: Because the `<summary>` element has a default role of [button](https://5042-brianpob-content-yemdwpko3pz.ws-us104.gitpod.io/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) (which strips all roles from child elements), this example will not work for users of assistive technologies such as screen readers. The `<h4>` will have its role removed and thus will not be treated as a heading for these users.

You have established that the "default role of button" statement is incorrect. And I believe the "this example will not work" is also not completely correct.

I did some testing using my Mac's built-in VoiceOver feature. While the screen reader did not recognize the `<h4>` in the example as a heading, as the warning mentions, it was still able to read the "Overview" text and I could still interact with the `<details>` element. 

## My question
How can this warning be made more accurate in terms of its accessibility-related behavior?

**Comment by scottaohara** [comment]
hi @Brian-Pob.  writing to acknowledge i've seen your ping.  I will come back to this asap.

File: files/en-us/web/html/element/summary/index.md [hunk 1]
```diff
@@ -127,11 +130,7 @@ This example adds some semantics to the `<summary>` element to indicate the labe
     <tr>
       <th scope="row">Implicit ARIA role</th>
       <td>
-        <code
-          ><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/button_role"
-            >button</a
-          ></code
-        >
+        <code>No corresponding role</code>
```
Comment (Mintoo200) [inline files/en-us/web/html/element/summary/index.md:133]: Should this still be in a `<code>` tag ?

Comment (Brian-Pob) [reply files/en-us/web/html/element/summary/index.md:133]: That's a good point. I missed that. I'll edit the code in the draft.

File: files/en-us/web/html/element/summary/index.md [hunk 2]
```diff
@@ -3,6 +3,9 @@ title: "<summary>: The Disclosure Summary element"
 slug: Web/HTML/Element/summary
 page-type: html-element
 browser-compat: html.elements.summary
+spec-urls:
+  - https://html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element
+  - https://w3c.github.io/html-aria/#el-summary
```
Comment (estelle) [inline files/en-us/web/html/element/summary/index.md:8]:
```suggestion
```
The spec url shows up correctly based on BCD. We don't generally include the aria spec in the HTML element page. This aria URL is just the role (or lack thereof) and links back to the whatwg link, so not including.

File: files/en-us/web/html/element/summary/index.md [hunk 3]
```diff
@@ -126,13 +126,7 @@ This example adds some semantics to the `<summary>` element to indicate the labe
...
     <tr>
       <th scope="row">Implicit ARIA role</th>
-      <td>
-        <code
-          ><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/button_role"
-            >button</a
-          ></code
-        >
-      </td>
+      <td>No corresponding role</td>
```
Comment (estelle) [inline files/en-us/web/html/element/summary/index.md:129]:
```suggestion
      <td> <a href="https://www.w3.org/TR/html-aria/#dfn-no-corresponding-role">No corresponding role</a></td>
```

**Review by estelle — APPROVED** [review]
Thanks! 🎉

**Comment by Brian-Pob** [comment]
@estelle Thanks for merging the PR, but it looks like I never actually updated the warning in the document. I updated the technical summary but not the warning.
Sorry about that.

Should I make a new PR?

Screenshot showing the unchanged warning for the summary element

**Pull request merged**